rust/linux-client/docs/use_cases.md

```diff
@@ -0,0 +1,42 @@
+# Linux client use cases
+
+This document lists the most common use cases that we could test. Exhaustively
+tesing everything would be difficult. These cases at least exercise every
+feature once, although they don't exercise every combination of features.
+
+1. Alpine Docker container, CLI only, resolv.conf DNS (Easy to test)
+2. Exe only, Manual startup, CLI only, No DNS (Embedded / IoT)
+3. Exe only, Manual startup, CLI only, resolv.conf DNS (Embedded / IoT)
+4. Package manager installation, activated by NetworkManager, CLI only, NetworkManager DNS (To get NM working before committing to a GUI)
+5. Package manager installation, runs as always-on systemd service unit, Tauri GUI, resolvectl DNS (A desktop that doesn't have NetworkManager)
+6. Package manager installation, runs as systemd service unit, activated by NetworkManager, Tauri GUI and NetworkManager interfaces, NetworkManager DNS (The "Make me one with everything" option)
```

Comment:
Does Tauri generate .deb and .rpm files for us? We probably won't have the resources to host package infra any time soon, but if we can build self-contained installers then great. Alternative is just providing a `.tar.gz` and instructions for unpacking.

Reply:
Since the binary is self-contained, it should rely on any dependencies, but also won't be configured to auto-updated.

We used to use Cloudsmith.io to host our old Firezone 0.7 packages. We could use them again.

Reply:
Yeah Tauri can build .deb and AppImage. Rpm, I'm not sure. `cargo-deb` can make debs for CLI exes. I think we should just make a `.deb` release artifact next to the client-x64 binaries on Github.

If we have any extra files like the systemd service unit and NetworkManager service file (for acting as NM plugin), it's easier to install, un-install, and upgrade them on Debian and Ubuntu than anything else.

e.g.
```
curl https://github.com/firezone/firezone/releases/whatever/firezone-client-x64.deb
sudo dpkg -i firezone-client-x64.deb
```

No apt repo needed.